src/api/app/models/repository.rb

```diff
@@ -61,6 +61,11 @@ def cleanup_before_destroy
     end
   end
 
+  def project_name
+    return self.project.name  if self.project
+    self.remote_project_name
+  end
```

Comment:
Yes, with the next condition you need to break it up again. While you can just insert another line with this writing .... matter of taste.